frontends/vmt_encoder.cpp

```diff
@@ -0,0 +1,68 @@
+/*********************                                                        */
+/*! \file vmt_encoder.cpp
+** \verbatim
+** Top contributors (to current version):
+**   Makai Mann
+** This file is part of the pono project.
+** Copyright (c) 2019 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved.  See the file LICENSE in the top-level source
+** directory for licensing information.\endverbatim
+**
+** \brief Frontend for the Verification Modulo Theories (VMT) format.
+**        See https://vmt-lib.fbk.eu/ for more information.
+**
+**
+**/
+
+#include "frontends/vmt_encoder.h"
+
+using namespace smt;
+using namespace std;
+
+namespace pono {
+
+VMTEncoder::VMTEncoder(std::string filename, RelationalTransitionSystem & rts)
+    : super(rts.get_solver()), filename_(filename), rts_(rts)
+{
+  set_logic_all();
+  int res = parse(filename_);
```

Comment:
I tried to run the example VMT model from the VMTLIB standard document (https://es-static.fbk.eu/people/griggio/papers/vmtlib.pdf).

```
; this is a comment
(declare-const x Int)
(declare-const x.next Int)
(define-fun sv.x () Int (! x :next x.next))
(declare-const b Bool)
(define-fun init () Bool
(! (= x 1) :init true))
(define-fun trans () Bool
(! (= x.next (ite b (+ x 1) x)) :trans true)
(define-fun p1 () Bool
(! (> x 0) :invar-property 1))
(define-fun p2 () Bool
(! (> x 10) :live-property 2))
```

The `parse` function fails with a non-zero exit code when running Pono with `--smt-solver=cvc4` to handle the `Int`. The last successful call of the `term_attribute` function is when handling the `:trans` keyword; `:next` and `:init` are handled correctly, as far as I can see.

Reply:
Thanks for that PR! It looks good to me overall, except for an assertion failure that is related to the `parse` function in the SmtLibReader (see above comment for details).

Reply:
Thanks for pointing that out! Looks like this is actually a benchmark issue, but it's too bad the error message isn't better. I think that would have to be a fix in smt-switch though. The `trans` `define-fun` is missing an ending parenthesis.